### ISSUE TITLE
No module named 'dust_extinction.averages'

### ISSUE THREAD

**Issue by kyika** [karllark/dust_extinction#84]
Hi,
I installed the package in a conda environment by first active that environment and then 
"pip install dust_extinction", the output of install process are print as follow:

(sci) ➜  ~ pip install dust_extinction
Collecting dust_extinction
  Downloading https://files.pythonhosted.org/packages/43/99/709405d47967296abd6b7f477231d533c85d516363825a9f06463fb4ffc8/dust_extinction-0.6.tar.gz (164kB)
    100% |████████████████████████████████| 174kB 76kB/s
Requirement already satisfied: astropy in /anaconda3/envs/sci/lib/python3.6/site-packages (from dust_extinction) (3.0.5)
Requirement already satisfied: scipy in /anaconda3/envs/sci/lib/python3.6/site-packages (from dust_extinction) (1.1.0)
Requirement already satisfied: numpy>=1.10.0 in /anaconda3/envs/sci/lib/python3.6/site-packages (from astropy->dust_extinction) (1.15.3)
Building wheels for collected packages: dust-extinction
  Running setup.py bdist_wheel for dust-extinction ... done
  Stored in directory: /Users/kyika/Library/Caches/pip/wheels/73/44/f7/b7629459b119cee574a89b01c1712bb7ef56bb77c1cd31687a
Successfully built dust-extinction
Installing collected packages: dust-extinction
Successfully installed dust-extinction-0.6
You are using pip version 10.0.1, however version 18.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

I think that it is successuful. However, when I using ipython or jupyter-notebook to import this package by:
from dust_extinction.averages import G03_SMCBar
I get an error:

ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-5-a184c01ce835> in <module>
----> 1 from dust_extinction.averages import G03_SMCBar

ModuleNotFoundError: No module named 'dust_extinction.averages'

What had happened here? Didn't I install a full-version of dust-extinction?

Any help and suggestion is grateful.

**Comment by karllark** [comment]
I do not understand why you are not able to find this module.   Are you just do the base import for averages?  Like 'import dust_extinction.averages'?

**Comment by karllark** [comment]
Trying to reproduce this in clean astroconda environment and I get the same error.  Standby, I am trying to understand the problem.

**Comment by karllark** [comment]
Ok, I now understand.  The version of the code that is on pip is v0.6.  For that version, all the code was in a single file called 'dust_extinction.py'.  The current development version has the separate modules for different types of extinction modules.  This was done to make it clearer the differences and make maintaining the code easier. 

Thus, the import statement you need is:

from dust_extinction.dust_extinction import G03_SMCBar

Note you can find the documentation for v0.6 on readthedocs.  Just switch from 'latest' to 'v0.6' using the floating button near the right corner of the page (on my browser).

**Comment by karllark** [comment]
There will be a new pip installable version soon that uses the new organization (e.g., 'averages' module instead of 'dust_extinction' module).  I do not expect that the organization of the code will change after v0.7.  See issue #82.

Apologies for the confusion.  Still learning the best way to create and maintain a python package others find useful.  :-)

**Comment by karllark** [comment]
If you want the latest version that will work with the 'averages' module, you can install directly from the github repository with:

pip install git+https://github.com/karllark/dust_extinction.git

**Comment by kyika** [comment]
Thanks,
I will try to install the latest version soon.

**Comment by kyika** [comment]
Hi,
It works for me. I can download and install the code in my astroconda environment, and calculate the curve in my code.

But the document page 
https://dust-extinction.readthedocs.io/en/latest/api/dust_extinction.averages.G03_SMCBar.html#dust_extinction.averages.G03_SMCBar
somewhat confuse me, I find that if I want to evaluate the Ax/Av in one wavelength(0.551micro, for example) , I just should write
x=1/0.551
ext_model(x)
rather than 
ext_model.evaluate(x)
actually, if I use the second method, I get an error: 
TypeError: object of type 'float' has no len()
and I find out that I have to give a array/list as parameter to the evaluate function

Though this cause very little confusion, I think it would be better if you can update the document and make it more clear.

**Comment by kyika** [comment]
One more question, have the curve been updated since the Gordon et al. 2003?
 I find the curve somewhat different from the formula in that paper.

**Comment by karllark** [comment]
Astropy models work a little different than you describe in the comment above.  The evaluate function is called implicitly after the model has been defined.  Hence,

ext_model = G03_SMCBar()  # define and initialize the model
x = 1/0.551
y = ext_model(x) # this command calls the evaluate member function

**Comment by karllark** [comment]
Yes.  In two ways.  First as noted in the code comments given below (will move this to the docs as well - see #86).  In addition, the data is smoothly interpolated in the optical/NIR and a FM90 functional form used for interpolation in the UV.

# values at 2.198 and 1.25 changed to provide smooth interpolation
# as noted in Gordon et al. (2016, ApJ, 826, 104)

**Comment by kyika** [comment]
Great, thanks for your reply. I now fully understand the doc of this package and find it ultra convenient! Thank you for share us your code!
